src/redistributors/morpho-dc-c/Distributor.sol

```diff
@@ -0,0 +1,118 @@
+
+    constructor(address wrappedMToken_, address morphoBlue_, address morphoVaultFactory_) {
+        wrappedMToken = wrappedMToken_;
+        morphoBlue = morphoBlue_;
+        morphoVaultFactory = morphoVaultFactory_;
+    }
+
+    /// @notice Allow `msg.sender` to deposit an amount of Morpho Vault shares to get access to WrappedM yield.
+    function deposit(address vault_, uint256 vaultShares_) external {
+        _revertIfNotMorphoVault(vault_);
+
+        if (!ERC20Helper.transferFrom(vault_, msg.sender, address(this), vaultShares_)) revert TransferFromFailed();
+
+        _addShares(_updateLastWrappedMBalance(), vault_, vaultShares_);
+        _addShares(vault_, _updateLastCumulativeDistribution(vault_), msg.sender, vaultShares_);
+    }
+
+    /// @notice Allow `msg.sender` to withdraw an amount of Morpho Vault shares to cease access to WrappedM yield.
+    function withdraw(address vault_, uint256 vaultShares_) external {
+        _revertIfNotMorphoVault(vault_);
+
+        _removeShares(_updateLastWrappedMBalance(), vault_, vaultShares_);
+        _removeShares(vault_, _updateLastCumulativeDistribution(vault_), msg.sender, vaultShares_);
+
+        if (!ERC20Helper.transfer(vault_, msg.sender, vaultShares_)) revert TransferFailed();
+    }
+
+    /// @notice Allow anyone to distribute any new WrappedM across all vaults.
+    function distribute() public {
+        _distribute(_updateLastWrappedMBalance());
+    }
+
+    /// @notice Allow anyone to distribute a vaults distributable WrappedM across all accounts.
+    function distribute(address vault_) public {
+        distribute();
+
+        _distribute(vault_, _updateLastCumulativeDistribution(vault_));
+    }
+
+    /// @notice Allow `msg.sender` to claim their WrappedM yield for their deposited shares of a Morpho Vault.
+    function claim(address vault_, address recipient_) external returns (uint256 claimed_) {
+        distribute(vault_);
+
+        IWrappedMLike(wrappedMToken).transfer(recipient_, claimed_ = _updateClaim(vault_, msg.sender));
+
+        _lastWrappedMBalance = IWrappedMLike(wrappedMToken).balanceOf(address(this));
+    }
+
+    /// @notice Returns the WrappedM yield claimable for an account's deposited shares of a Morpho Vault.
+    function getClaimable(address vault_, address account_) public view returns (uint256 claimable_) {
+        // NOTE: Should/could never be negative.
+        return _getCumulativeDistribution(vault_, account_) - _claims[vault_][account_];
+    }
+
+    /// @notice Returns an account's deposited shares of a Morpho Vault.
+    function getDeposits(address vault_, address account_) external view returns (uint256 shares_) {
+        return _getShares(vault_, account_);
+    }
+
+    /// @notice Returns an account's shares of distribution of a vault's distribution.
+    function getShares(address vault_, address account_) external view returns (uint256 shares_) {
+        return _getShares(vault_, account_);
+    }
```

Comment:
Yes, since this version of the model assumes that all deposited vault shares are always equivalent in how many wM assets they represent, then a accounts shares of a vault distribution is also their deposits. I did it this way to:
- allow a comparison between the 3 models, so you can see the implications for each external function
- prepare for future use where `getDeposits` will not return the same thing as `getShares` where share prices are variable